### ISSUE TITLE
Unable to click on the elements with Protractor 4.0.12

### ISSUE THREAD

**Issue by ZaidMomin1** [angular/protractor#3803]
Hi,

Updated protractor version to 4.0.12 and all the tests started failing, giving Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL. error message. 

 Node Version: 4.4.7
 Protractor Version: 
 Angular Version: 1.0.6
Jasmine version: 2.4.1
Browser name: chrome
Browser version: 55.0.2883.75
Platform: Windows NT

Below is the package.json snippet: 
scripts": {
        "test": "protractor ./conf.js",
        "pretest": "npm run update-webdriver",
        "update-webdriver": "webdriver-manager update",
        "preupdate-webdriver": "npm install"
    },
"dependencies": {
        "colors": "^1.1.2",
        "express": "~3.4.0",
        "firefox-profile": "0.3.4",
        "protractor": "4.0.12",
         "jasmine-spec-reporter": "^2.7.0",
        "protractor-jasmine2-screenshot-reporter": "latest",
        "request": "latest"
}

Test Code:
Spec File 
describe('QualityCheck-ItemSearch', function() {
    beforeEach(function() {
        itemSearchPage.clickQualityCheckinMenuBar();
    });
    it('should navigate to product details page on entering correct itemId.', function() {
        itemSearchPage.searchItem(itemDetails.validOrderItemId);
     });
});

ItemSearchPage File
var ItemSearchPage = function() {
    this.searchItem = function(orderItemId) {
        helper.enterInputInTextBox(this.searchTextbox, helper.LOCATE_BY.MODEL, orderItemId);
         helper.clickElement(this.searchButton, helper.LOCATE_BY.ID);
    };
});
module.exports = new ItemSearchPage();

Helper File
var Helper = function() {
    this.clickElement = function(locator_name, locate_by) {
        switch (locate_by) {
            case 'id':
                element(by.id(locator_name)).click();
                break;
            case 'model':
                element(by.model(locator_name)).click();
                break;
       }
    }
};
module.exports = new Helper();

Output of execution:
Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL. 

In-order to resolve the issue, I just happen to degrade the protractor version to 4.0.11 and tests ran successfully.

Could you please look into the issue and address the same. 

Thanks,
Zaid

**Comment by NickTomlin** [comment]
Could you please follow the [issue template](https://github.com/angular/protractor/blob/master/.github/ISSUE_TEMPLATE)? That will make it easier for us to help you. Thanks!

**Comment by isian8814** [comment]
After update to 4.0.12, parallel tests starts failing due to the elements not clickable. 
```
[chrome #01-2]   Message:
[chrome #01-2]     Failed: unknown error: Element is not clickable at point (1117, 921). Other element would receive the click: <ul id="sortable_records" class="ui-sortable">...</ul>
[chrome #01-2]       (Session info: chrome=54.0.2840.71)
[chrome #01-2]       (Driver info: chromedriver=2.25.426924 (649f9b868f6783ec9de71c123212b908bf3b232e),platform=Linux 4.4.27-moby x86_64) (WARNING: The server did not provide any stacktrace information)
[chrome #01-2]     Command duration or timeout: 418 milliseconds
[chrome #01-2]     Build info: version: '3.0.1', revision: '1969d75', time: '2016-10-18 09:48:19 -0700'
[chrome #01-2]     System info: host: 'c09e94e8bcd5', ip: '172.17.0.7', os.name: 'Linux', os.arch: 'amd64', os.version: '4.4.27-moby', java.version: '1.8.0_91'
[chrome #01-2]     Driver info: org.openqa.selenium.chrome.ChromeDriver
[chrome #01-2]     Capabilities [{applicationCacheEnabled=false, rotatable=false, mobileEmulationEnabled=false, networkConnectionEnabled=false, chrome={chromedriverVersion=2.25.426924 (649f9b868f6783ec9de71c123212b908bf3b232e), userDataDir=/tmp/.org.chromium.Chromium.LU6v1H}, takesHeapSnapshot=true, pageLoadStrategy=normal, databaseEnabled=false, handlesAlerts=true, hasTouchScreen=false, version=54.0.2840.71, platform=LINUX, browserConnectionEnabled=false, nativeEvents=true, acceptSslCerts=true, locationContextEnabled=true, webStorageEnabled=true, browserName=chrome, takesScreenshot=true, javascriptEnabled=true, cssSelectorsEnabled=true}]
[chrome #01-2]     Session ID: dc3324e968330e348f6df36b801f5cc5
[chrome #01-2]   Stack:
[chrome #01-2]     WebDriverError: unknown error: Element is not clickable at point (1117, 921). Other element would receive the click: <ul id="sortable_records" class="ui-sortable">...</ul>
[chrome #01-2]       (Session info: chrome=54.0.2840.71)
[chrome #01-2]       (Driver info: chromedriver=2.25.426924 (649f9b868f6783ec9de71c123212b908bf3b232e),platform=Linux 4.4.27-moby x86_64) (WARNING: The server did not provide any stacktrace information)
[chrome #01-2]     Command duration or timeout: 418 milliseconds
[chrome #01-2]     Build info: version: '3.0.1', revision: '1969d75', time: '2016-10-18 09:48:19 -0700'
[chrome #01-2]     System info: host: 'c09e94e8bcd5', ip: '172.17.0.7', os.name: 'Linux', os.arch: 'amd64', os.version: '4.4.27-moby', java.version: '1.8.0_91'
[chrome #01-2]     Driver info: org.openqa.selenium.chrome.ChromeDriver
[chrome #01-2]     Capabilities [{applicationCacheEnabled=false, rotatable=false, mobileEmulationEnabled=false, networkConnectionEnabled=false, chrome={chromedriverVersion=2.25.426924 (649f9b868f6783ec9de71c123212b908bf3b232e), userDataDir=/tmp/.org.chromium.Chromium.LU6v1H}, takesHeapSnapshot=true, pageLoadStrategy=normal, databaseEnabled=false, handlesAlerts=true, hasTouchScreen=false, version=54.0.2840.71, platform=LINUX, browserConnectionEnabled=false, nativeEvents=true, acceptSslCerts=true, locationContextEnabled=true, webStorageEnabled=true, browserName=chrome, takesScreenshot=true, javascriptEnabled=true, cssSelectorsEnabled=true}]
[chrome #01-2]     Session ID: dc3324e968330e348f6df36b801f5cc5
[chrome #01-2]         at WebDriverError (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/error.js:27:5)
[chrome #01-2]         at Object.checkLegacyResponse (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/error.js:639:15)
[chrome #01-2]         at parseHttpResponse (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/http/index.js:538:13)
[chrome #01-2]         at client_.send.then.response (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/http/index.js:472:11)
[chrome #01-2]         at ManagedPromise.invokeCallback_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:1379:14)
[chrome #01-2]         at TaskQueue.execute_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2913:14)
[chrome #01-2]         at TaskQueue.executeNext_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2896:21)
[chrome #01-2]         at asyncRun (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2820:25)
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:639:7
[chrome #01-2]         at process._tickCallback (internal/process/next_tick.js:103:7)
[chrome #01-2]     From: Task: WebElement.click()
[chrome #01-2]         at WebDriver.schedule (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/webdriver.js:377:17)
[chrome #01-2]         at WebElementPromise.schedule_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/webdriver.js:1744:25)
[chrome #01-2]         at WebElementPromise.click (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/webdriver.js:1845:17)
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/test/js/biz/neustar/ultra/commons/CommonUIMethods.js:149:53
[chrome #01-2]         at ManagedPromise.invokeCallback_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:1379:14)
[chrome #01-2]         at TaskQueue.execute_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2913:14)
[chrome #01-2]         at TaskQueue.executeNext_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2896:21)
[chrome #01-2]         at asyncRun (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2775:27)
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:639:7
[chrome #01-2]         at process._tickCallback (internal/process/next_tick.js:103:7)
[chrome #01-2]     From: Task: Run beforeEach in control flow
[chrome #01-2]         at Object.<anonymous> (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/jasminewd2/index.js:79:14)
[chrome #01-2]     From asynchronous test: 
[chrome #01-2]     Error
[chrome #01-2]         at Suite.<anonymous> (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/test/js/biz/neustar/ultra/geoip/specs/DefineNewGroup/addDirectionalPoolRecordOfTypeMX.js:70:5)
[chrome #01-2]         at Object.<anonymous> (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/test/js/biz/neustar/ultra/geoip/specs/DefineNewGroup/addDirectionalPoolRecordOfTypeMX.js:18:1)
[chrome #01-2]         at Module._compile (module.js:556:32)
[chrome #01-2]         at Object.Module._extensions..js (module.js:565:10)
[chrome #01-2]         at Module.load (module.js:473:32)
[chrome #01-2]         at tryModuleLoad (module.js:432:12)
[chrome #01-2]   Message:
[chrome #01-2]     Failed: Error while waiting for Protractor to sync with the page: "window.angular is undefined.  This could be either because this is a non-angular page or because your test involves client-side navigation, which can interfere with Protractor's bootstrapping.  See http://git.io/v4gXM for details"
[chrome #01-2]   Stack:
[chrome #01-2]     Error: Error while waiting for Protractor to sync with the page: "window.angular is undefined.  This could be either because this is a non-angular page or because your test involves client-side navigation, which can interfere with Protractor's bootstrapping.  See http://git.io/v4gXM for details"
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/protractor/built/browser.js:271:23
[chrome #01-2]         at ManagedPromise.invokeCallback_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:1379:14)
[chrome #01-2]         at TaskQueue.execute_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2913:14)
[chrome #01-2]         at TaskQueue.executeNext_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2896:21)
[chrome #01-2]         at asyncRun (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2775:27)
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:639:7
[chrome #01-2]         at process._tickCallback (internal/process/next_tick.js:103:7)Error
[chrome #01-2]         at ElementArrayFinder.applyAction_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/protractor/built/element.js:460:27)
[chrome #01-2]         at ElementArrayFinder._this.(anonymous function) [as isDisplayed] (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/protractor/built/element.js:104:30)
[chrome #01-2]         at ElementFinder.(anonymous function) [as isDisplayed] (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/protractor/built/element.js:826:22)
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/test/js/biz/neustar/ultra/commons/CommonUIMethods.js:358:37
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/webdriver.js:716:14
[chrome #01-2]         at TaskQueue.execute_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2913:14)
[chrome #01-2]         at TaskQueue.executeNext_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2896:21)
[chrome #01-2]         at asyncRun (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2820:25)
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:639:7
[chrome #01-2]         at process._tickCallback (internal/process/next_tick.js:103:7)
[chrome #01-2]     From: Task: <anonymous>
[chrome #01-2]         at pollCondition (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2359:16)
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2355:9
[chrome #01-2]         at new ManagedPromise (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:1082:7)
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2354:14
[chrome #01-2]         at TaskQueue.execute_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2913:14)
[chrome #01-2]         at TaskQueue.executeNext_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2896:21)
[chrome #01-2]         at asyncRun (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2820:25)
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:639:7
[chrome #01-2]         at process._tickCallback (internal/process/next_tick.js:103:7)
[chrome #01-2]     From: Task: The element is not displayed - by.model("main.poolInfo.description")
[chrome #01-2]         at ControlFlow.wait (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2352:17)
[chrome #01-2]         at WebDriver.wait (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/webdriver.js:712:29)
[chrome #01-2]         at Object.inputTextAfterElementDisplay (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/test/js/biz/neustar/ultra/commons/CommonUIMethods.js:357:24)
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/test/js/biz/neustar/ultra/pageMethods/DirectionalPoolPageMethods.js:104:29
[chrome #01-2]         at TaskQueue.execute_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2913:14)
[chrome #01-2]         at TaskQueue.executeNext_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2896:21)
[chrome #01-2]         at asyncRun (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2820:25)
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:639:7
[chrome #01-2]         at process._tickCallback (internal/process/next_tick.js:103:7)
[chrome #01-2]     From: Task: <anonymous>
[chrome #01-2]         at Object.inputDirectionalPoolInfo (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/test/js/biz/neustar/ultra/pageMethods/DirectionalPoolPageMethods.js:103:21)
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/test/js/biz/neustar/ultra/geoip/specs/DefineNewGroup/addDirectionalPoolRecordOfTypeMX.js:938:44
[chrome #01-2]         at ManagedPromise.invokeCallback_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:1379:14)
[chrome #01-2]         at TaskQueue.execute_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2913:14)
[chrome #01-2]         at TaskQueue.executeNext_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2896:21)
[chrome #01-2]         at asyncRun (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2775:27)
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:639:7
[chrome #01-2]         at process._tickCallback (internal/process/next_tick.js:103:7)
[chrome #01-2]     From: Task: Run it("Create Directional Pool Records") in control flow
[chrome #01-2]         at Object.<anonymous> (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/jasminewd2/index.js:79:14)
[chrome #01-2]         at /Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/jasminewd2/index.js:103:16
[chrome #01-2]         at ManagedPromise.invokeCallback_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:1379:14)
[chrome #01-2]         at TaskQueue.execute_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2913:14)
[chrome #01-2]         at TaskQueue.executeNext_ (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/node_modules/selenium-webdriver/lib/promise.js:2896:21)
[chrome #01-2]     From asynchronous test: 
[chrome #01-2]     Error
[chrome #01-2]         at Suite.<anonymous> (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/test/js/biz/neustar/ultra/geoip/specs/DefineNewGroup/addDirectionalPoolRecordOfTypeMX.js:922:5)
[chrome #01-2]         at Object.<anonymous> (/Users/minchanglong/Documents/SourceTree/java_ui_blackbox_test/test/js/biz/neustar/ultra/geoip/specs/DefineNewGroup/addDirectionalPoolRecordOfTypeMX.js:18:1)
[chrome #01-2]         at Module._compile (module.js:556:32)
[chrome #01-2]         at Object.Module._extensions..js (module.js:565:10)
[chrome #01-2]         at Module.load (module.js:473:32)
[chrome #01-2]         at tryModuleLoad (module.js:432:12)
```

**Comment by NickTomlin** [comment]
@isian8814 could you please add some additional info per the [issue template](https://github.com/angular/protractor/blob/master/.github/ISSUE_TEMPLATE) as well? 

It would be wonderful to have a small test case that we can use to reproduce and fix this.

**Comment by cnishina** [comment]
The stacktrace is unhelpful since your tests are in parallel. I am not sure which of the several errors your bug report is referring to.

**Comment by cnishina** [comment]
Please try again with 4.0.13. Thanks!

**Comment by ZaidMomin1** [comment]
@NickTomlin have updated the issue according to the template.

**Comment by NickTomlin** [comment]
@ZaidMomin1 thanks! Could you try again with protractor `4.0.13`

**Comment by NickTomlin** [comment]
I'm going to go ahead and close this as it should be fixed by `4.0.13`

**Comment by ZaidMomin1** [comment]
Hi @NickTomlin,

Update the protractor version to 4.0.13 and it's work as expected. That is the issue of protractor not able to click on the elements on ui is resolved. 

Thanks,
Zaid
